rdt/transformers/pii/anonymizer.py

```diff
@@ -209,3 +214,148 @@ def __repr__(self):
 
         args_string = ', '.join(custom_args)
         return f'{class_name}({args_string})'
+
+
+class PseudoAnonymizedFaker(AnonymizedFaker):
+    """Pseudo-anonymization Transformer using Faker.
+
+    This transformer anonymizes values that can be traced back to the original input by using
+    a mapping. The transformer will generate a mapping with the previously specified
+    ``Faker`` provider and ``function``.
+
+    Args:
+        provider_name (str):
+            The name of the provider in ``Faker``. If ``None`` the ``BaseProvider`` is used.
+            Defaults to ``None``.
+        function_name (str):
+            The name of the function to use within the ``faker.provider``. Defaults to
+            ``lexify``.
+        function_kwargs (dict):
+            Keyword args to pass into the ``function_name`` when being called.
+        locales (list):
+            List of localized providers to use instead of the global provider.
+    """
+
+    def __getstate__(self):
+        """Return a dictionary representation of the instance and warn the user when pickling."""
+        LOGGER.warning((
+            'You are saving the mapping information, which includes the original data. '
+            'Sharing this object with others will also give them access to the original data '
+            'used with this transformer.'
+        ))
+
+        return self.__dict__
+
+    def __init__(self, provider_name=None, function_name=None, function_kwargs=None, locales=None):
+        super().__init__(
+            provider_name=provider_name,
+            function_name=function_name,
+            function_kwargs=function_kwargs,
+            locales=locales,
+        )
+
+        self._mapping_dict = {}
+        self._reverse_mapping_dict = {}
+        self._label_encoder = None
+
+    def get_mapping(self):
+        """Return the mapping dictionary."""
+        return deepcopy(self._mapping_dict)
+
+    def get_output_sdtypes(self):
+        """Return the output sdtypes supported by the transformer.
+
+        Returns:
+            dict:
+                Mapping from the transformed column names to supported sdtypes.
+        """
+        output_sdtypes = {'value': 'float'}
```

Comment:
Can we just set `self.OUTPUT_SDTYPES` to this dict? The base class has a method that does the next line already